corehq/apps/accounting/management/commands/list_customer_billing_account_software_plan.py

```diff
@@ -0,0 +1,94 @@
+from django.core.management.base import BaseCommand
+from corehq.apps.accounting.models import BillingAccount, Subscription, SubscriptionType
+import csv
+
+
+class Command(BaseCommand):
+    # Always run this command without '--update' first
+    # let Ops review the output before running with '--update'
+    help = 'List customer billing accounts and associated software plans'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--update', action='store_true', default=False,
+                            help='Update all software plans (type PRODUCT or IMPLEMENTATION) '
+                            'under each billing account to the main billing subscription')
+
+    def handle(self, *args, **kwargs):
+        allow_update = kwargs['update']
+        customer_billing_accounts = []
+        while True:
+            account_name = input("Enter a customer billing account name, "
+                                 "type 'ALL' for all customer billing account, "
+                                 "type 'DONE' when finished:\n")
+            if account_name.upper() == 'ALL':
+                customer_billing_accounts = ['ALL']
+                break
+            elif account_name.upper() == 'DONE':
+                break
+            else:
+                try:
+                    account = BillingAccount.objects.get(name=account_name,
+                                                         is_active=True, is_customer_billing_account=True)
+                except BillingAccount.DoesNotExist:
+                    print(f"Customer billing account {account_name} does not exist")
+                else:
+                    customer_billing_accounts.append(account)
+
+        if 'ALL' in customer_billing_accounts:
+            customer_billing_accounts = BillingAccount.objects.filter(
+                is_customer_billing_account=True, is_active=True)
```

Comment:
Looks like `customer_billing_accounts` can contain two different types of objects (string or account). This is a common cause of bugs. Luckily I think this one can be simplified by moving the condition below into this `if` block.
```suggestion
                customer_billing_accounts = BillingAccount.objects.filter(
                    is_customer_billing_account=True, is_active=True)
                break
            elif account_name.upper() == 'DONE':
                break
            else:
                try:
                    account = BillingAccount.objects.get(name=account_name,
                                                         is_active=True, is_customer_billing_account=True)
                except BillingAccount.DoesNotExist:
                    print(f"Customer billing account {account_name} does not exist")
                else:
                    customer_billing_accounts.append(account)
```